invenio_records_resources/factories/factory.py

```diff
@@ -124,11 +133,18 @@ def create_metadata_model(self):
 
     def create_record_class(self):
         """Create record class."""
+        pid_field = PIDField(
```

Comment:
Is it worth creating a `PIDField` factory, and just make the full field configurable rather than passing many parameters? This time are 3, maybe later on are more... @lnielsen

Reply:
I agree......i see two options. 1) override the entire pidfield or 2) have two parameters:

```
pid_provider_cls
pid_field_kwargs
```

Reply:
I opted by the two args to:
1- Keep consistency on the naming (i.e. have a `_cls`)
2- Avoid possible problems on lazyness/partial because it would need an instance instead of a class created at config level.